subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt

```diff
@@ -17,37 +17,97 @@
 #
 
 ##############################################################################
-##
-##  ${applicationName} start up script for UN*X
-##
+#
+#   Gradle start up script for POSIX
+#
+#   You're probably reading this because you're a Unix/Linux person and have
+#   been instructed to run "gradlew" to install or build a project that's of
+#   interest to you. (At least, that's how I wound up reading this script.)
+#
+#   Here's some stuff you need to know.
+#
+#   (0) You need a POSIX-compliant shell to run this script. If your /bin/sh
+#       is noncompliant, but you have some other shell that is (perhaps ksh or
+#       bash) then to run this script, type the shell name before the whole
+#       command line, like:
+#
+#           ksh gradlew [...args...]
+#
+#   (1) Even if you have a compliant shell, Java is fussy about $PWD, so you
+#       might as well cd to the directory containing gradlew and run it as
+#
+#           ./gradlew
+#
+#   (2) Gradle is a build system used by the project that you're building or
+#       installing. It's like "autoconf+make" but for projects that are written
+#       in Java.
+#
+#   (3) Gradle is mostly written in Java, but there's wrapper scripts for
+#       various environments, including this one. You can find it at
+#       https://github.com/gradle/gradle
+#
+#   (4) Because we're targeting any POSIX shell, we progressively modify "$@",
+#       rather than try and built up all the args just before invoking Java.
+#       We even pass shellcheck.
+#
+#   (5) This script (gradlew) is generated from
+#       subprojects/plugins/src/main/resources/org/gradle/api/internal/plugins/unixStartScript.txt
+#       in the gradle project, so submitting patches will be tedious.
```

Comment:
As I said above, this script is used for things beyond `gradlew`, so it probably doesn't make sense to include these details that may only apply to Gradle or builds.

It's sort of useful to know that this is a generated file and it was generated by Gradle

Reply:
I've re-organized this.

My objective is to provide a starting point for someone like myself, who's a seasoned Unix shell user, but who hasn't done Java programming in a modern framework. (The last Java program I wrote targetted Java version 1.4, and everything was done on the command line, and I had a Makefile for building jars.)

When the instructions say _run_ this script, the first thing I do is _read_ the script.

So I'm including what to you must seem like a great deal of redundant information, but it's not redundant to the target audience.